src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http3.cs

```diff
@@ -1677,13 +1684,31 @@ public async Task ServerClosesOutboundControlStream_ClientClosesConnection(bool
                 await using Http3LoopbackStream requestStream = await connection.AcceptRequestStreamAsync();
 
                 // abort the control stream
-                if (graceful)
+                if (closeType == CloseOutboundControlStream.BogusData)
                 {
                     await connection.OutboundControlStream.SendResponseBodyAsync(Array.Empty<byte>(), isFinal: true);
                 }
-                else
+                else if (closeType == CloseOutboundControlStream.Dispose)
                 {
-                    connection.OutboundControlStream.Abort(Http3LoopbackConnection.H3_INTERNAL_ERROR);
+                    await connection.OutboundControlStream.DisposeAsync();
+                }
+                else if (closeType == CloseOutboundControlStream.Abort)
+                {
+                    int iterations = 5;
+                    while (iterations-- > 0)
```

Comment:
Depending on timing, the client can receive the control stream abort, i.e. RESET_STREAM, before it even read the HTTP/3 stream type from it, see https://github.com/dotnet/runtime/blob/f69470a91ac823962f3dcb919f8a92f4bb85bde1/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs#L459 for details.

QUIC spec says, RESET can discard data, which can happen in this case. Add to it H/3 spec, which says that unrecognized or aborted stream should be ignored:
https://github.com/dotnet/runtime/blob/f69470a91ac823962f3dcb919f8a92f4bb85bde1/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs#L477-L485
Leading to ignoring the incoming control stream, because client doesn't know it's control, instead of reacting to the abort and closing the connection. 

So this fix attempts to do the again, but with slight delay between sending the data and abort, line 1710.


And why this worked before: we would return data for one QuicStream.ReadAsync before propagating the error, not the whole buffer, just one call to read succeed, which was enough to get that 1 byte saying it's control. Also, MsQuic could always discard the data, we just never observed it (timing, too fast on the same machine...)